keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/DoubleIfVertex.java

```diff
@@ -13,24 +16,51 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public class DoubleIfVertex extends DoubleVertex implements Differentiable, NonProbabilistic<DoubleTensor> {
+import static io.improbable.keanu.tensor.TensorShapeValidation.checkHasSingleNonScalarShapeOrAllScalar;
+
+public class DoubleIfVertex extends DoubleVertex implements SaveableVertex, Differentiable, NonProbabilistic<DoubleTensor> {
 
     private final Vertex<? extends BooleanTensor> predicate;
-    private final Vertex<? extends DoubleTensor> thn;
-    private final Vertex<? extends DoubleTensor> els;
+    private final DoubleVertex thn;
+    private final DoubleVertex els;
+    protected static final String PREDICATE_NAME = "predicate";
+    protected static final String THN_NAME = "thn";
```

Comment:
Can we just call these `THEN_NAME` and `ELSE_NAME` and do `getThen` etc. The `thn` and `els` are just to avoid the keyword

Reply:
or do they need to exactly match the params?

Reply:
No - they don't need to match to work.  I'll rename.